core/serialization/blocks.js

```diff
@@ -427,6 +449,28 @@ const tryToConnectParent = function(parentConnection, child, state) {
   }
 };
 
+/**
+ * Applies icon state to the icons on the block, based on the given state
+ * object.
+ * @param {!Block} block The block to set the icon state of.
+ * @param {!State} state The state object to reference.
+ */
+const loadIcons = function(block, state) {
+  if (!state['icons']) {
+    return;
+  }
+  // TODO(#2105): Remove this logic and put it in the icon.
+  const comment = state['icons']['comment'];
+  if (comment) {
+    block.setCommentText(comment['text']);
+    block.commentModel.pinned = comment['pinned'];
+    block.commentModel.size = new Size(comment['width'], comment['height']);
+    if (comment['pinned'] && block.getCommentIcon && !block.isInFlyout) {
+      setTimeout(() => block.getCommentIcon().setVisible(true), 1);
```

Comment:
As it happens, yes! But only because [git is wonderful](https://github.com/google/blockly/blame/0726e4a909ed6875b1066b75f0b8b96dd8a58bd4/core/xml.js#L661). (I just stole this from the XML system, and the comment there got removed at some point hehe.)